Merged Parachain repo

zktony opened the pull request:
## Describe your changes
Merged Parachain repo.
- [x] Fix try-runtime error.
- [x] Generate new weights.

zktony committed:
Merged parachain and solochain

Gauthamastro commented on nodes/polkadex-mainnet-node/Cargo.toml:
rename the folders as mainnet and parachain, it's cleaner.

Gauthamastro commented on runtimes/polkadex-mainnet/Cargo.toml:
Here too, mainnet and parachain. its cleaner

zktony committed:
Upgraded depend

zktony committed:
Fixed build

Gauthamastro committed:
Install pallets

zktony committed:
Install Swap pallet (#872)

zktony committed:
Fixed tests

zktony committed:
Upgraded dependencies to 1.1.0 (#871)

## Describe your changes
Upgraded dependencies to 1.1.0.

zktony committed:
Merge branch 'Develop' into zk-reorg

serhii-temchenko commented on .docker/ci/release/Dockerfile:
```diff
@@ -22,11 +22,11 @@ RUN apt-get update && apt-get install --assume-yes ca-certificates
-    mkdir -p /polkadex-node/.local/share && \
+RUN useradd -m -u 1000 -U -s /bin/sh -d /polkadex-polkadex-mainnet-polkadex-parachain-node polkadex-polkadex-mainnet-polkadex-parachain-node && \
+    mkdir -p /polkadex-polkadex-mainnet-polkadex-parachain-node/.local/share && \
     mkdir /data && \
-    chown -R polkadex-node:polkadex-node /data && \
-    ln -s /data /polkadex-node/.local/share/polkadex-node && \
+    chown -R polkadex-polkadex-mainnet-polkadex-parachain-node:polkadex-polkadex-mainnet-polkadex-parachain-node /data && \
+    ln -s /data /polkadex-polkadex-mainnet-polkadex-parachain-node/.local/share/polkadex-polkadex-mainnet-polkadex-parachain-node && \
```
Is this some global rename missed here?  I'm about `polkadex-polkadex-mainnet-polkadex-parachain-node` names.

zktony replied:
Yeah .. these files got messed.

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -86,14 +86,14 @@ jobs:
-        run: cargo build -p node-polkadex-runtime
-      - name: Build Project with runtime benchmarks
-        run: cargo build --features runtime-benchmarks
+        run: cargo build -p polkadex-mainnet-polkadex-parachain-node-polkadex-polkadex-mainnet
```
Same

zktony replied:
Same reason

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -86,14 +86,14 @@ jobs:
+        run: cargo build --features polkadex-mainnet-benchmarks
       - name: Test weights generation
-        run: ./target/debug/polkadex-node benchmark pallet --pallet "*" --extrinsic "*" --steps 2 --repeat 1
+        run: ./target/debug/polkadex-polkadex-mainnet-polkadex-parachain-node benchmark pallet --pallet "*" --extrinsic "*" --steps 2 --repeat 1
```
Same

zktony replied:
same reason

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -110,7 +110,7 @@ jobs:
         with:
           version: '0.22.0'
-          args: '--avoid-cfg-tarpaulin --exclude polkadex-node node-polkadex-runtime --workspace --timeout 180'
+          args: '--avoid-cfg-tarpaulin --exclude polkadex-polkadex-mainnet-polkadex-parachain-node polkadex-mainnet-polkadex-parachain-node-polkadex-polkadex-mainnet --workspace --timeout 180'
```
Same

zktony replied:
same reason

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -125,13 +125,13 @@ jobs:
         if: contains(github.ref, 'Develop')
         # Zipping is required since billing is based on the raw uploaded size.
-        run: zip -r -j polkadex-node.zip ./target/release/polkadex-node
+        run: zip -r -j polkadex-polkadex-mainnet-polkadex-parachain-node.zip ./target/release/polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

zktony replied:
same reason

serhii-temchenko commented on .github/workflows/ci.yml:
```diff
@@ -125,13 +125,13 @@ jobs:
-          name: polkadex-node
-          path: ./polkadex-node.zip
+          name: polkadex-polkadex-mainnet-polkadex-parachain-node
+          path: ./polkadex-polkadex-mainnet-polkadex-parachain-node.zip
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -16,24 +16,24 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
-# Docker compose file to start a multi node local test network.
+# Docker compose file to start a multi polkadex-mainnet-polkadex-parachain-node local test network.
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -16,24 +16,24 @@
 # 1. Build `target/release/substrate` binary: `cargo build --release`
 # 2. Start networks and containers:
-#    `sudo docker-compose -f .maintain/sentry-node/docker-compose.yml up`
+#    `sudo docker-compose -f .maintain/sentry-polkadex-mainnet-polkadex-parachain-node/docker-compose.yml up`
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -50,12 +50,12 @@ services:
       - "9615:9615"
     volumes:
-      - ../../target/release/polkadex-node:/usr/local/bin/polkadex-node
+      - ../../target/release/polkadex-polkadex-mainnet-polkadex-parachain-node:/usr/local/bin/polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -50,12 +50,12 @@ services:
       - internet
     command:
-      - "--node-key"
+      - "--polkadex-mainnet-polkadex-parachain-node-key"
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -81,11 +81,11 @@ services:
       - "9945:9944"
     volumes:
-      - ../../target/release/polkadex-node:/usr/local/bin/polkadex-node
+      - ../../target/release/polkadex-polkadex-mainnet-polkadex-parachain-node:/usr/local/bin/polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -81,11 +81,11 @@ services:
       - internet
     command:
-      - "--node-key"
+      - "--polkadex-mainnet-polkadex-parachain-node-key"
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -111,11 +111,11 @@ services:
       - "9946:9944"
     volumes:
-      - ../../target/release/polkadex-node:/usr/local/bin/polkadex-node
+      - ../../target/release/polkadex-polkadex-mainnet-polkadex-parachain-node:/usr/local/bin/polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on .maintain/local-docker-test-network/docker-compose.yml:
```diff
@@ -111,11 +111,11 @@ services:
       - internet
     command:
-      - "--node-key"
+      - "--polkadex-mainnet-polkadex-parachain-node-key"
```
Same

serhii-temchenko commented on README.md:
```diff
@@ -41,7 +41,7 @@ Note that cloning master might result in an unstable build. If you want a stable
 cd Polkadex
 
-# Build the node (The first build will be long (~30min))
+# Build the polkadex-mainnet-polkadex-parachain-node (The first build will be long (~30min))
```
Same

serhii-temchenko commented on README.md:
````diff
@@ -54,15 +54,15 @@ source $HOME/.cargo/env
 
 ```bash
-./target/release/polkadex-node --dev
+./target/release/polkadex-polkadex-mainnet-polkadex-parachain-node --dev
````
Same

serhii-temchenko commented on README.md:
````diff
@@ -54,15 +54,15 @@ source $HOME/.cargo/env
 
 ```bash
-./target/release/node-polkadex purge-chain --dev
+./target/release/polkadex-mainnet-polkadex-parachain-node-polkadex purge-chain --dev
````
Same

serhii-temchenko commented on README.md:
```diff
@@ -84,16 +84,16 @@ Alice_Node_Key=$(subkey generate --scheme Ed25519 --output-type Json | jq -r '.s
   --ws-port 9945 \
   --rpc-port 9933 \
-  --node-key $Alice_Node_Key \
+  --polkadex-mainnet-polkadex-parachain-node-key $Alice_Node_Key \
```
Same

serhii-temchenko commented on README.md:
```diff
@@ -106,13 +106,13 @@ Alice_Node_Key=$(subkey generate --scheme Ed25519 --output-type Json | jq -r '.s
 2021-06-30 08:12:39 👶 Creating empty BABE epoch changes on what appears to be first startup.    
 2021-06-30 08:12:39 Using default protocol ID "sup" because none is configured in the chain specs    
-2021-06-30 08:12:39 🏷 Local node identity is: 12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp    
+2021-06-30 08:12:39 🏷 Local polkadex-mainnet-polkadex-parachain-node identity is: 12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp    
```
Same

serhii-temchenko commented on README.md:
```diff
@@ -126,8 +126,8 @@ Local node identity is: 12D3KooWEyoppNCUx8Yx66oV9fJnriXwCcXwDDUA2kj6vnc6iDEp sho
-./target/release/polkadex-node purge-chain --base-path /tmp/bob --chain local
-./target/release/polkadex-node \
+./target/release/polkadex-polkadex-mainnet-polkadex-parachain-node purge-chain --base-path /tmp/bob --chain local
+./target/release/polkadex-polkadex-mainnet-polkadex-parachain-node \
```
Same

serhii-temchenko commented on README.md:
```diff
@@ -149,18 +149,18 @@ If all is going well, after a few seconds, the nodes should peer together and st
-2021-06-30 08:16:53 🔍 Discovered new external address for our node: /ip4/127.0.0.1/tcp/30334/p2p/12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
-2021-06-30 08:16:53 🔍 Discovered new external address for our node: /ip4/192.168.1.37/tcp/30334/p2p/12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
+2021-06-30 08:16:53 🔍 Discovered new external address for our polkadex-mainnet-polkadex-parachain-node: /ip4/127.0.0.1/tcp/30334/p2p/12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
+2021-06-30 08:16:53 🔍 Discovered new external address for our polkadex-mainnet-polkadex-parachain-node: /ip4/192.168.1.37/tcp/30334/p2p/12D3KooWRHDuuHg5ZQcJhvVDKud9XkFz2Dcs2GQKF9KKuTD6quq7    
```
Same

serhii-temchenko commented on README.md:
````diff
@@ -180,7 +180,7 @@ If all is going well, after a few seconds, the nodes should peer together and st
 
 ```bash
-docker build . -t polkadex-node
+docker build . -t polkadex-polkadex-mainnet-polkadex-parachain-node
````
Same

serhii-temchenko commented on dockerfile:
```diff
@@ -35,11 +35,11 @@ RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y && \
-        chown -R polkadex-node:polkadex-node /data && \
-        ln -s /data /polkadex-node/.local/share/polkadex-node && \
+        chown -R polkadex-polkadex-mainnet-polkadex-parachain-node:polkadex-polkadex-mainnet-polkadex-parachain-node /data && \
+        ln -s /data /polkadex-polkadex-mainnet-polkadex-parachain-node/.local/share/polkadex-polkadex-mainnet-polkadex-parachain-node && \
```
Same

serhii-temchenko commented on nodes/polkadex-mainnet-node/Cargo.toml:
Shouldn't all dependencies be taken from the workspace?

zktony replied:
We were having issues with node dependencies. I guess it's because of the default feature.

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/cli.rs:
```diff
@@ -54,18 +54,19 @@ pub enum Subcommand {
+	// about = "Decode given block or extrinsic using current native polkadex-mainnet."
 	// )]
 	// Inspect(node_inspect::cli::InspectCmd),
-	/// The custom benchmark subcommmand benchmarking runtime pallets.
+	/// The custom benchmark subcommmand benchmarking polkadex-mainnet pallets.
 	#[clap(subcommand)]
 	Benchmark(Box<frame_benchmarking_cli::BenchmarkCmd>),
 
-	/// Try some command against runtime state.
+	/// Try some command against polkadex-mainnet state.
 	#[cfg(feature = "try-runtime")]
 	TryRuntime(try_runtime_cli::TryRuntimeCmd),
 
-	/// Try some command against runtime state. Note: `try-runtime` feature must be enabled.
+	/// Try some command against polkadex-mainnet state. Note: `try-polkadex-mainnet` feature must
+	/// be enabled.
```
Are these changes intentional?

zktony replied:
yeah .. this is fine

serhii-temchenko replied:
But there is no feature like `try-polkadex-mainnet`.

zktony replied:
I reverted it. It looks like something happened during the revert.

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -104,10 +104,10 @@ pub fn run() -> Result<()> {
 							return Err(
-								"Runtime benchmarking wasn't enabled when building the node. \
-							You can enable it with `--features runtime-benchmarks`."
+								"Runtime benchmarking wasn't enabled when building the polkadex-mainnet-polkadex-parachain-node. \
```
Same with `polkadex-mainnet-polkadex-parachain-node` name

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -104,10 +104,10 @@ pub fn run() -> Result<()> {
-								"Runtime benchmarking wasn't enabled when building the node. \
-							You can enable it with `--features runtime-benchmarks`."
+								"Runtime benchmarking wasn't enabled when building the polkadex-mainnet-polkadex-parachain-node. \
+							You can enable it with `--features polkadex-mainnet-benchmarks`."
```
There is no feature with this name

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -104,10 +104,10 @@ pub fn run() -> Result<()> {
 				match cmd.as_ref() {
 					BenchmarkCmd::Pallet(cmd) => {
-						if !cfg!(feature = "runtime-benchmarks") {
+						if !cfg!(feature = "polkadex-mainnet-benchmarks") {
```
There is no feature with this name

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -120,7 +120,7 @@ pub fn run() -> Result<()> {
 					#[cfg(not(feature = "runtime-benchmarks"))]
 					BenchmarkCmd::Storage(_) => Err(
-						"Storage benchmarking can be enabled with `--features runtime-benchmarks`."
+						"Storage benchmarking can be enabled with `--features polkadex-mainnet-benchmarks`."
```
There is no feature with this name

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -229,7 +229,7 @@ pub fn run() -> Result<()> {
 			let runner = cli.create_runner(cmd)?;
 			runner.async_run(|config| {
-				// we don't need any of the components of new_partial, just a runtime, or a task
+				// we don't need any of the components of new_partial, just a polkadex-mainnet, or a task
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -250,8 +250,8 @@ pub fn run() -> Result<()> {
-		Some(Subcommand::TryRuntime) => Err("TryRuntime wasn't enabled when building the node. \
-        You can enable it with `--features try-runtime`."
+		Some(Subcommand::TryRuntime) => Err("TryRuntime wasn't enabled when building the polkadex-mainnet-polkadex-parachain-node. \
+        You can enable it with `--features try-polkadex-mainnet`."
```
There is no feature with this name

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/command.rs:
```diff
@@ -250,8 +250,8 @@ pub fn run() -> Result<()> {
 		#[cfg(not(feature = "try-runtime"))]
-		Some(Subcommand::TryRuntime) => Err("TryRuntime wasn't enabled when building the node. \
-        You can enable it with `--features try-runtime`."
+		Some(Subcommand::TryRuntime) => Err("TryRuntime wasn't enabled when building the polkadex-mainnet-polkadex-parachain-node. \
```
Look like IDE bulk change. Please confirm that this is valid update

zktony replied:
yeah .. I did a folder name refactor. It looks like it updated the comments also.

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/rpc.rs:
Same with changes in this file. Please check all of them and confirm that this is valid update

serhii-temchenko commented on nodes/polkadex-mainnet-node/src/service.rs:
Same with changes in this file. Please check all of them and confirm that this is valid update

serhii-temchenko commented on nodes/polkadex-parachain-node/Cargo.toml:
Shouldn't all dependencies be taken from the workspace?

zktony replied:
Workspace dependencies cause problems for nodes. I guess we tried it earlier and it didn't work.

serhii-temchenko commented on nodes/polkadex-parachain-node/build.rs:
Please add some documentation and license

serhii-temchenko commented on nodes/polkadex-parachain-node/src/chain_spec.rs:
Please add some documentation and license

serhii-temchenko commented on nodes/polkadex-parachain-node/src/cli.rs:
Please add some documentation and license

serhii-temchenko commented on nodes/polkadex-parachain-node/src/command.rs:
Please add some documentation and license

serhii-temchenko commented on nodes/polkadex-parachain-node/src/main.rs:
Please add license block

serhii-temchenko commented on nodes/polkadex-parachain-node/src/rpc.rs:
Please add a license block

serhii-temchenko commented on nodes/polkadex-parachain-node/src/service.rs:
Please add a license block

serhii-temchenko commented on pallets/liquidity/src/lib.rs:
```diff
@@ -48,7 +48,7 @@ type BalanceOf<T> =
 const LENGTH_OF_HALF_BYTES: usize = 16;
 
-// Definition of the pallet logic, to be aggregated at runtime definition through
+// Definition of the pallet logic, to be aggregated at polkadex-mainnet definition through
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/liquidity/src/mock.rs:
```diff
@@ -29,27 +29,27 @@ use frame_system::{EnsureRoot, EnsureSigned};
 type Block = frame_system::mocking::MockBlock<Test>;
 
-// For testing the pallet, we construct a mock runtime.
+// For testing the pallet, we construct a mock polkadex-mainnet.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/liquidity/src/weights.rs:
```diff
@@ -7,7 +7,7 @@
 
 // Executed Command:
-// ./polkadex-node
+// ./polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on pallets/ocex/rpc/src/lib.rs:
```diff
@@ -67,10 +67,10 @@ pub trait PolkadexOcexRpcApi<BlockHash, AccountId, Hash> {
-/// * `Client`: The client API used to interact with the Substrate runtime.
-/// * `Block`: The block type of the Substrate runtime.
+/// * `Client`: The client API used to interact with the Substrate polkadex-mainnet.
+/// * `Block`: The block type of the Substrate polkadex-mainnet.
 pub struct PolkadexOcexRpc<Client, Block, T: OffchainStorage + 'static> {
-	/// An `Arc` reference to the client API for accessing runtime functionality.
+	/// An `Arc` reference to the client API for accessing polkadex-mainnet functionality.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/ocex/rpc/src/lib.rs:
```diff
@@ -114,8 +114,9 @@ where
-		// WARN: this is a hack on beating the boundry of runtime -> node
-		// with decoding tuple of underlying data into solid std type
+		// WARN: this is a hack on beating the boundry of polkadex-mainnet ->
+		// polkadex-mainnet-polkadex-parachain-node with decoding tuple of underlying data into
```
Same

serhii-temchenko commented on pallets/ocex/rpc/src/lib.rs:
```diff
@@ -197,7 +198,7 @@ where
 	}
 }
 
-/// Converts a runtime trap into an RPC error.
+/// Converts a polkadex-mainnet trap into an RPC error.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/ocex/src/lib.rs:
```diff
@@ -126,7 +127,7 @@ pub trait OcexWeightInfo {
 	fn whitelist_orderbook_operator() -> Weight;
 }
 
-// Definition of the pallet logic, to be aggregated at runtime definition through
+// Definition of the pallet logic, to be aggregated at polkadex-mainnet definition through
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/ocex/src/storage.rs:
```diff
@@ -220,7 +220,8 @@ impl HashDB<BlakeTwo256, DBValue> for State {
 }
 
-/// Derive a database key from hash value of the node (key) and  the node prefix.
+/// Derive a database key from hash value of the polkadex-mainnet-polkadex-parachain-node (key) and
+/// the polkadex-mainnet-polkadex-parachain-node prefix.
```
Same

serhii-temchenko commented on pallets/ocex/src/weights.rs:
```diff
@@ -7,7 +7,7 @@
 
 // Executed Command:
-// ./polkadex-node
+// ./polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on pallets/pdex-migration/src/lib.rs:
```diff
@@ -71,14 +71,15 @@ pub mod pallet {
 	/// Configure the pallet by specifying the parameters and types on which it depends.
 	pub trait Config: frame_system::Config + pallet_balances::Config + pallet_sudo::Config {
-		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		/// Because this pallet emits events, it depends on the polkadex-mainnet's definition of an
+		/// event.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/pdex-migration/src/mock.rs:
```diff
@@ -21,29 +21,26 @@ use frame_system as system;
 type Block = frame_system::mocking::MockBlock<Test>;
 type Balance = u128;
-// Configure a mock runtime to test the pallet.
+// Configure a mock polkadex-mainnet to test the pallet.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/pdex-migration/src/weights.rs:
```diff
@@ -8,7 +8,7 @@
 
 // Executed Command:
-// ./polkadex-node
+// ./polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on pallets/rewards/rpc/src/lib.rs:
```diff
@@ -49,10 +49,10 @@ pub trait PolkadexRewardsRpcApi<BlockHash, AccountId, Hash> {
-/// * `Client`: The client API used to interact with the Substrate runtime.
-/// * `Block`: The block type of the Substrate runtime.
+/// * `Client`: The client API used to interact with the Substrate polkadex-mainnet.
+/// * `Block`: The block type of the Substrate polkadex-mainnet.
 pub struct PolkadexRewardsRpc<Client, Block> {
-	/// An `Arc` reference to the client API for accessing runtime functionality.
+	/// An `Arc` reference to the client API for accessing polkadex-mainnet functionality.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/rewards/src/lib.rs:
```diff
@@ -66,7 +66,7 @@ pub mod weights;
 const MIN_REWARDS_CLAIMABLE_AMOUNT: u128 = polkadex_primitives::UNIT_BALANCE;
 pub const REWARDS_LOCK_ID: LockIdentifier = *b"REWARDID";
-// Definition of the pallet logic, to be aggregated at runtime definition through
+// Definition of the pallet logic, to be aggregated at polkadex-mainnet definition through
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea-council/Cargo.toml:
```diff
@@ -0,0 +1,60 @@
+targets = ["x86_64-unknown-linux-gnu"]
+
+[dependencies]
+codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
+scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
```
Could these be used from the workspace?

serhii-temchenko commented on pallets/thea-executor/src/lib.rs:
```diff
@@ -48,7 +48,7 @@ pub trait WeightInfo {
 	use super::*;
 	use frame_support::{
-		log,
+		__private::log,
```
What is this `__private`? is this a mistake?

Gauthamastro replied:
@zktony Please update here.

serhii-temchenko commented on pallets/thea-executor/src/lib.rs:
```diff
@@ -70,7 +70,8 @@ pub mod pallet {
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
-		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		/// Because this pallet emits events, it depends on the polkadex-mainnet's definition of an
+		/// event.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea-executor/src/mock.rs:
```diff
@@ -176,8 +174,8 @@ where
 }
 
-// Build genesis storage according to the mock runtime.
+// Build genesis storage according to the mock polkadex-mainnet.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea-executor/src/weights.rs:
```diff
@@ -7,7 +7,7 @@
 
 // Executed Command:
-// ./polkadex-node
+// ./polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on pallets/thea-message-handler/src/mock.rs:
```diff
@@ -23,25 +23,26 @@ use sp_core::H256;
 type Block = frame_system::mocking::MockBlock<Test>;
 type Balance = u128;
 
-// Configure a mock runtime to test the pallet.
+// Configure a mock polkadex-mainnet to test the pallet.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea-message-handler/src/mock.rs:
```diff
@@ -184,8 +184,8 @@ where
 }
 
-// Build genesis storage according to the mock runtime.
+// Build genesis storage according to the mock polkadex-mainnet.
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea-message-handler/src/weights.rs:
```diff
@@ -7,7 +7,7 @@
 
 // Executed Command:
-// ./polkadex-node
+// ./polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on pallets/thea/src/lib.rs:
```diff
@@ -21,9 +21,9 @@
 //!
-//! The core logic for runtime is handled by the Thea pallet. The most important
+//! The core logic for polkadex-mainnet is handled by the Thea pallet. The most important
 //! responsibilities of the Thea pallet are to:
-//! * process ingress messages to the runtime;
+//! * process ingress messages to the polkadex-mainnet;
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea/src/session.rs:
```diff
@@ -33,8 +33,8 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 		let authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
-		// we panic here as runtime maintainers can simply reconfigure genesis and restart the
-		// chain easily
+		// we panic here as polkadex-mainnet maintainers can simply reconfigure genesis and restart
+		// the chain easily
```
Look like IDE bulk change. Please confirm that this is valid update

serhii-temchenko commented on pallets/thea/src/weights.rs:
```diff
@@ -7,7 +7,7 @@
 
 // Executed Command:
-// ./polkadex-node
+// ./polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko commented on rpc/assets/src/lib.rs:
```diff
@@ -58,8 +58,8 @@ pub trait PolkadexAssetHandlerRpcApi<BlockHash, AccountId, Hash> {
 ///
-/// * `Client`: The client API used to interact with the Substrate runtime.
-/// * `Block`: The block type of the Substrate runtime.
+/// * `Client`: The client API used to interact with the Substrate polkadex-mainnet.
+/// * `Block`: The block type of the Substrate polkadex-mainnet.
```
Same

serhii-temchenko commented on rpc/assets/src/lib.rs:
```diff
@@ -103,7 +103,7 @@ where
 }
 
-/// Converts a runtime trap into an RPC error.
+/// Converts a polkadex-mainnet trap into an RPC error.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/Cargo.toml:
```diff
@@ -78,10 +82,10 @@ frame-election-provider-support = { default-features = false, workspace = true }
 pallet-assets = { default-features = false, workspace = true }
 #end added
 
-# Used for the node template"s RPCs
+# Used for the polkadex-mainnet-polkadex-parachain-node template"s RPCs
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/Cargo.toml:
```diff
@@ -243,7 +250,7 @@ try-runtime = [
   "pdex-migration/try-runtime",
   "pallet-democracy/try-runtime",
   "pallet-preimage/try-runtime",
-  #"pallet-randomness-collective-flip/try-runtime",
+  #"pallet-randomness-collective-flip/try-polkadex-mainnet",
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/constants.rs:
```diff
@@ -15,7 +15,7 @@
 // limitations under the License.
 
-//! A set of constant values used in substrate runtime.
+//! A set of constant values used in substrate polkadex-mainnet.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/impls.rs:
```diff
@@ -15,7 +15,7 @@
 // limitations under the License.
 
-//! Some configurable implementations as associated type for the substrate runtime.
+//! Some configurable implementations as associated type for the substrate polkadex-mainnet.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/impls.rs:
```diff
@@ -65,7 +65,7 @@ mod multiplier_tests {
 	}
 
-	// update based on runtime impl.
+	// update based on polkadex-mainnet impl.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/impls.rs:
```diff
@@ -158,7 +158,7 @@ mod multiplier_tests {
 	#[test]
 	fn time_to_reach_zero() {
-		// blocks per 24h in substrate-node: 28,800 (k)
+		// blocks per 24h in substrate-polkadex-mainnet-polkadex-parachain-node: 28,800 (k)
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -16,38 +16,41 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! The Substrate runtime. This can be compiled with `#[no_std]`, ready for Wasm.
+//! The Substrate polkadex-mainnet. This can be compiled with `#[no_std]`, ready for Wasm.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -88,10 +91,10 @@ use sp_version::NativeVersion;
 use sp_version::RuntimeVersion;
 use static_assertions::const_assert;
 
-/// Implementations of some helper traits passed into runtime modules as associated types.
+/// Implementations of some helper traits passed into polkadex-mainnet modules as associated types.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -88,10 +91,10 @@ use sp_version::NativeVersion;
 
-/// Implementations of some helper traits passed into runtime modules as associated types.
+/// Implementations of some helper traits passed into polkadex-mainnet modules as associated types.
 pub mod impls;
 
-/// Constant values used within the runtime.
+/// Constant values used within the polkadex-mainnet.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -110,11 +113,11 @@ pub fn wasm_binary_unwrap() -> &'static [u8] {
 
 /// Runtime version.
 pub const VERSION: RuntimeVersion = RuntimeVersion {
-	spec_name: create_runtime_str!("node"),
+	spec_name: create_runtime_str!("polkadex-mainnet-polkadex-parachain-node"),
 	impl_name: create_runtime_str!("polkadex-official"),
 	authoring_version: 10,
-	// Per convention: if the runtime behavior changes, increment spec_version
-	// and set impl_version to 0. If only runtime
+	// Per convention: if the polkadex-mainnet behavior changes, increment spec_version
+	// and set impl_version to 0. If only polkadex-mainnet
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -1358,124 +1363,166 @@ impl thea_message_handler::Config for Runtime {
 		//RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Pallet, Storage} = 32,
 		ChildBounties: pallet_child_bounties = 33,
-		Assets: pallet_assets::{Pallet, Call, Storage, Event<T>} = 34,
-		OCEX: pallet_ocex_lmp::{Pallet, Call, Storage, Event<T>, ValidateUnsigned} = 35,
-		OrderbookCommittee: pallet_collective::<Instance3>::{Pallet, Call, Storage, Origin<T>, Event<T>} = 36,
-		Thea: thea::pallet::{Pallet, Call, Storage, Event<T>,ValidateUnsigned} = 39,
-		Rewards: pallet_rewards::{Pallet, Call, Storage, Event<T>} = 40,
-		Liquidity: liquidity::{Pallet, Call, Storage, Event<T>} = 41,
-		TheaExecutor: thea_executor::pallet::{Pallet, Call, Storage, Event<T>} = 44,
+		Assets: pallet_assets = 34,
+		OCEX: pallet_ocex_lmp = 35,
+		OrderbookCommittee: pallet_collective::<Instance3> = 36,
+		Thea: thea::pallet = 39,
+		Rewards: pallet_rewards = 40,
+		Liquidity: liquidity = 41,
+		TheaExecutor: thea_executor::pallet = 44,
+		AssetConversion: pallet_asset_conversion = 46,
+		AssetConversionTxPayment: pallet_asset_conversion_tx_payment = 47,
 	}
 );
 /// Digest item type.
 pub type DigestItem = generic::DigestItem;
 /// The address format for describing accounts.
 pub type Address = sp_runtime::MultiAddress<AccountId, AccountIndex>;
-/// Block header type as expected by this runtime.
+/// Block header type as expected by this polkadex-mainnet.
 pub type Header = generic::Header<BlockNumber, BlakeTwo256>;
-/// Block type as expected by this runtime.
+/// Block type as expected by this polkadex-mainnet.
 pub type Block = generic::Block<Header, UncheckedExtrinsic>;
 /// A Block signed with a Justification
 pub type SignedBlock = generic::SignedBlock<Block>;
-/// BlockId type as expected by this runtime.
+/// BlockId type as expected by this polkadex-mainnet.
 pub type BlockId = generic::BlockId<Block>;
 /// The SignedExtension to the basic transaction logic.
 ///
-/// When you change this, you **MUST** modify [`sign`] in `bin/node/testing/src/keyring.rs`!
+/// When you change this, you **MUST** modify [`sign`] in
+/// `bin/polkadex-mainnet-polkadex-parachain-node/testing/src/keyring.rs`!
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -1485,9 +1532,9 @@ pub type SignedExtra = (
 	frame_system::CheckMortality<Runtime>,
 	frame_system::CheckNonce<Runtime>,
 	frame_system::CheckWeight<Runtime>,
-	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
+	pallet_asset_conversion_tx_payment::ChargeAssetTxPayment<Runtime>,
 );
-/// Unchecked extrinsic type as expected by this runtime.
+/// Unchecked extrinsic type as expected by this polkadex-mainnet.
```
Same

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -1726,7 +1793,7 @@ impl_runtime_apis! {
 	}
 
 
-	#[cfg(feature = "try-runtime")]
+	#[cfg(feature = "try-polkadex-mainnet")]
```
I don't see this feature

serhii-temchenko commented on runtimes/polkadex-mainnet/src/lib.rs:
```diff
@@ -1748,7 +1815,7 @@ impl_runtime_apis! {
 		}
 	}
 
-	#[cfg(feature = "runtime-benchmarks")]
+	#[cfg(feature = "polkadex-mainnet-benchmarks")]
```
Same

serhii-temchenko commented on runtimes/polkadex-parachain/build.rs:
Please add docs and license block

serhii-temchenko commented on runtimes/polkadex-parachain/src/lib.rs:
Please add docks and license block

serhii-temchenko commented on scripts/start_chain.sh:
```diff
@@ -24,7 +24,7 @@ start_boot_node() {
   install -d ../ind_validators/validator1
   cd ../ind_validators/validator1
-  ../../target/$TARGET/polkadex-node --validator --base-path ./bootnode -lthea=trace -lorderbook=trace --rpc-port=9943 --thea-dummy-mode --chain=../../scripts/customSpecRaw.json --node-key=1f64f01767da8258fcb986bd68d6dff93dfcd49d0fc753cea27cf37ce91c3684 >out_boot_node 2>&1 &
+  ../../target/$TARGET/polkadex-polkadex-mainnet-polkadex-parachain-node --validator --base-path ./bootnode -lthea=trace -lorderbook=trace --rpc-port=9943 --thea-dummy-mode --chain=../../scripts/customSpecRaw.json --polkadex-mainnet-polkadex-parachain-node-key=1f64f01767da8258fcb986bd68d6dff93dfcd49d0fc753cea27cf37ce91c3684 >out_boot_node 2>&1 &
```
Same

zktony replied:
same reason

serhii-temchenko commented on scripts/start_chain.sh:
```diff
@@ -33,11 +33,11 @@ start_validator_1() {
     --bootnodes /ip4/127.0.0.1/tcp/30333/p2p/12D3KooWRozCnsH7zCYiNVpCRqgaoxukPdYxqaPQNs9rdDMDeN4t \
     --bootnodes /ip4/127.0.0.1/tcp/30335/p2p/12D3KooWCMKvu1tJKQBjDZ4hN1saTP6D58e4WkwLZwks5cPpxqY7 \
-    --node-key=d353c4b01db05aa66ddeab9d85c2fa2252368dd4961606e5985ed1e8f40dbc50 >out_validator_1 2>&1 &
+    --polkadex-mainnet-polkadex-parachain-node-key=d353c4b01db05aa66ddeab9d85c2fa2252368dd4961606e5985ed1e8f40dbc50 >out_validator_1 2>&1 &
```
Same

serhii-temchenko commented on scripts/start_chain.sh:
```diff
@@ -46,11 +46,11 @@ start_validator_2() {
     --bootnodes /ip4/127.0.0.1/tcp/30333/p2p/12D3KooWRozCnsH7zCYiNVpCRqgaoxukPdYxqaPQNs9rdDMDeN4t \
     --bootnodes /ip4/127.0.0.1/tcp/30334/p2p/12D3KooWEVBdwVmV1BeAdtqzhjANK31ibYmLQXxEoeai4fx7KhNh \
-    --node-key=24f121a84149f784f9fe3f1e2fb04e8873191a510bc4b073a3a815d78a29cf2d >out_validator_2 2>&1 &
+    --polkadex-mainnet-polkadex-parachain-node-key=24f121a84149f784f9fe3f1e2fb04e8873191a510bc4b073a3a815d78a29cf2d >out_validator_2 2>&1 &
```
Same

serhii-temchenko commented on scripts/start_chain.sh:
```diff
@@ -59,7 +59,7 @@ start_full_node() {
   install -d ../ind_validators/full_node
   cd ../ind_validators/full_node
-  ../../target/$TARGET/polkadex-node  --chain=../../scripts/customSpecRaw.json --pruning=archive \
+  ../../target/$TARGET/polkadex-polkadex-mainnet-polkadex-parachain-node  --chain=../../scripts/customSpecRaw.json --pruning=archive \
```
Same

serhii-temchenko commented on scripts/start_chain.sh:
```diff
@@ -68,7 +68,7 @@ start_full_node() {
 kill_nodes() {
   echo "Killing nodes..."
-  killall -2 polkadex-node
+  killall -2 polkadex-polkadex-mainnet-polkadex-parachain-node
```
Same

serhii-temchenko reviewed (changes requested):
Plus one question -  since bumped version of the code libraries - shouldn't weights be updated?

zktony committed:
Updated weights

serhii-temchenko committed:
fix(cs): Toml files code style fix

serhii-temchenko committed:
feat(containerization): Fixed dockerfiles

serhii-temchenko committed:
feat(ci): Integrated parachain node build and check

serhii-temchenko committed:
feat(cd): Integrated parachain node build and artifact upload to the release

zktony committed:
Fixed polkadoex node errors

zktony committed:
Fixed rutime issue

Gauthamastro committed:
Fixed polkadex-node

Gauthamastro committed:
Remove TODO

zktony committed:
Reverted name-change

serhii-temchenko committed:
Merge branch 'zk-reorg' into feat/no-ref/ci-fix

# Conflicts:
#	dockerfile
#	runtimes/mainnet/Cargo.toml

serhii-temchenko committed:
feat(no-ref): Code style fix

serhii-temchenko commented:
Changed label to `InProgress` since changes from PR description section are required. Please change it back when those points will be implemented.

zktony committed:
Fixed runtime

serhii-temchenko committed:
Merge branch 'zk-reorg' into feat/no-ref/ci-fix

# Conflicts:
#	.github/workflows/ci.yml
#	runtimes/mainnet/Cargo.toml

serhii-temchenko commented on nodes/mainnet/src/cli.rs:
```diff
@@ -61,9 +61,10 @@ pub enum Subcommand {
-	/// Try some command against runtime state.
-	#[cfg(feature = "try-runtime")]
-	TryRuntime(try_runtime_cli::TryRuntimeCmd),
+	/// Try-runtime has migrated to a standalone
+	/// [CLI](<https://github.com/paritytech/try-runtime-cli>). The subcommand exists as a stub and
+	/// deprecation notice. It will be removed entirely some time after Janurary 2024.
+	TryRuntime,
```
On the line 71 we already have this property. Please remove one. It fails to compile because of this.

serhii-temchenko commented on nodes/mainnet/src/lib.rs:
```diff
@@ -22,3 +22,4 @@ pub mod chain_spec;
 pub mod cli;
 pub mod node_rpc;
 pub mod service;
+use try_runtime_cli as _;
```
Is it required or just missed here? Not featuregated which leads to compilation error.

serhii-temchenko commented on nodes/parachain/src/cli.rs:
```diff
@@ -54,21 +53,31 @@ pub enum Subcommand {
-	subcommand_negates_reqs = true
+propagate_version = true,
+args_conflicts_with_subcommands = true,
+subcommand_negates_reqs = true
```
Please add tabulation back.

serhii-temchenko commented on nodes/parachain/src/cli.rs:
```diff
@@ -118,4 +127,4 @@ impl RelayChainCli {
 		}
 	}
-}
+}
```
Please add empty line back.

zktony committed:
Fixed benchmarks

zktony committed:
Added host function

zktony committed:
Updated conf

zktony committed:
Added tabulation

zktony committed:
Added line

serhii-temchenko committed:
Merge branch 'zk-reorg' into feat/no-ref/ci-fix

# Conflicts:
#	runtimes/mainnet/Cargo.toml

serhii-temchenko committed:
Code style fixes

serhii-temchenko committed:
chore(build): Fixed containerization for local test network

serhii-temchenko committed:
chore(cs): Fixed code style and removed unused imports and members

serhii-temchenko committed:
chore(build): Fixed node container running issue by adding required dependency

serhii-temchenko commented:
Currently there are couple issues which leads to build fails and multiple formatting/naming problems which resolved in PR #873. That PR couldn't be merged because of credentials issue related to the CI and described in [this comment](https://github.com/Polkadex-Substrate/Polkadex/pull/873#issuecomment-1771993652).

serhii-temchenko commented:
About weights generation - we have a delay because of devops team (running and giving access to the server). Assist requested 4 days ago, still waiting on access...

serhii-temchenko committed:
feat(it-241): Updated pallets weight after substrate version bump

serhii-temchenko committed:
chore(cs): Removed unused imports

zktony committed:
Fixed benchmarks

serhii-temchenko committed:
Merge branch 'zk-reorg' into feat/no-ref/ci-fix

# Conflicts:
#	runtimes/parachain/Cargo.toml

Gauthamastro commented on nodes/mainnet/src/service.rs:
```diff
@@ -373,13 +390,6 @@ pub fn new_full_base(
 			warp_sync_params: Some(WarpSyncParams::WithProvider(warp_sync)),
 		})?;
-	// Enable offchain indexing
-	sc_service::build_offchain_workers(
```
Why was this removed?

Gauthamastro replied:
Removed

Gauthamastro commented on runtimes/mainnet/src/lib.rs:
```diff
@@ -85,6 +89,7 @@ use sp_runtime::{
 use sp_std::{prelude::*, vec};
 #[cfg(any(feature = "std", test))]
 use sp_version::NativeVersion;
+use sp_storage as _;
```
Can be removed.

serhii-temchenko committed:
chore(cleanup): Removed file added by mistake

Gauthamastro committed:
Fix Comments

Gauthamastro committed:
Merge branch 'Develop' into zk-reorg

serhii-temchenko committed:
Merge branch 'zk-reorg' into feat/no-ref/ci-fix

# Conflicts:
#	runtimes/mainnet/Cargo.toml
#	runtimes/mainnet/src/impls.rs

zktony committed:
Added missing impl

serhii-temchenko committed:
Merge branch 'zk-reorg' into feat/no-ref/ci-fix

# Conflicts:
#	runtimes/parachain/Cargo.toml

serhii-temchenko committed:
chore(cs): Code style fixes

zktony committed:
Fixed benchmarks

zktony committed:
fixed linter error

zktony committed:
Fixed warning

zktony committed:
Fixed formatting

zktony committed:
Updated test case

serhii-temchenko committed:
chore(cs): Code style fixes

serhii-temchenko committed:
chore(cs): Cleaned up commented out code

serhii-temchenko committed:
fix(ci): Fixed issue with no space on self hosted runner

serhii-temchenko committed:
fix(ci): Increased self hosted runner storage size

Gauthamastro committed:
Modify storage size

Gauthamastro committed:
Modify AMI

Gauthamastro committed:
New AMI with 500 GB

Gauthamastro committed:
New AMI with 500 GB

Gauthamastro committed:
Update Fee Multiplier

serhii-temchenko committed:
Merge branch 'feat/no-ref/ci-fix' into feat/it-241/weights

serhii-temchenko committed:
feat(it-241): Weights update

serhii-temchenko committed:
chore(cs): Removed outdated comment

serhii-temchenko committed:
Merge branch 'feat/no-ref/ci-fix' into feat/it-241/weights

serhii-temchenko committed:
feat(it-241): Updated parachain related pallets weights

Gauthamastro committed:
Updated pallets weight (#876)

## Describe your changes
After substrate version update in PR #870 weights of pallets should be
updated.
In this PR newly generated weights are provided. 

## Issue ticket number and link
[#241](https://github.com/Polkadex-Substrate/issue-tracker/issues/241)

## Checklist before requesting a review
- [X] I have performed a self-review of my code.
- [X] If it is a core feature, I have added thorough tests.
- [X] I removed all Clippy and Formatting Warnings. 
- [X] I added required Copyrights.

~~:warning: Note that parachain related pallets weights (e.g.
`thea-council` and `xcm-helper`) didn't updated because of error:~~

> ~~Error: Input("Did not find the benchmarking metadata. This could
mean that you either did not build the node correctly with the
`--features runtime-benchmarks` flag, or the chain spec that you are
using was not created by a node that was compiled with the flag: Other:
Exported method Benchmark_benchmark_metadata is not found")~~

~~Even though `--features runtime-benchmarks` flag provided on the build
process.
@zktony please check this problem.~~ Fixed.

Gauthamastro committed:
Cargo update
